src/menu.rs

```diff
@@ -32,67 +41,63 @@ impl Menubar {
   }
 }
 
-// menu builder
+/// Base `Menu` functions.
 #[derive(Debug, Clone)]
 pub struct Menu {
   pub(crate) menu_platform: MenuPlatform,
   pub(crate) menu_type: MenuType,
 }
 
 impl Menu {
+  /// Creates a new Menu for Menubar/Window context.
   pub fn new(menu_type: MenuType) -> Self {
     Self {
       menu_platform: MenuPlatform::new(),
       menu_type,
     }
   }
 
+  /// Creates a new Menu for Popup context.
   pub fn new_popup_menu(menu_type: MenuType) -> Self {
     Self {
       menu_platform: MenuPlatform::new_popup_menu(),
       menu_type,
     }
   }
 
+  /// Shortcut to add a submenu to this `Menu`.
+  ///
+  /// This function is equivalent to [`Menu::add_item(self, MenuItem::Submenu(title, enabled, submenu))`].
+  /// See [`Menu::add_item`] for details.
+  ///
+  /// [`Menu::add_item`]: crate::menu::Menu
   pub fn add_submenu(&mut self, title: &str, enabled: bool, submenu: Menu) {
     self.menu_platform.add_item(
       MenuItem::Submenu(title.to_string(), enabled, submenu.menu_platform),
       self.menu_type,
     );
   }
 
-  pub fn add_custom_item(
-    &mut self,
-    text: &str,
-    keyboard_accelerator: Option<&str>,
-    enabled: bool,
-    selected: bool,
-  ) -> (MenuId, CustomMenuItem) {
-    let menu_id = MenuId::new(&text);
-    let item = self.menu_platform.add_custom_item(
-      menu_id,
-      self.menu_type,
-      text,
-      keyboard_accelerator,
-      enabled,
-      selected,
-    );
-    (menu_id, item)
-  }
-
+  /// Add new item to this menu.
   pub fn add_item(&mut self, item: MenuItem) -> Option<CustomMenuItem> {
-    self.menu_platform.add_item(item, self.menu_type)
+    self.menu_platform.add_item(item.clone(), self.menu_type)
   }
 }
 
-/// A menu item, bound to a pre-defined action or `Custom` emit an event. Note that status bar only
-/// supports `Custom` menu item variants. And on the menu bar, some platforms might not support some
-/// of the variants. Unsupported variant will be no-op on such platform.
+/// A menu item, bound to a pre-defined action or `Custom` emit an event. Note some platforms
+/// might not support some of the variants. Unsupported variant will be no-op on such platform.
```

Comment:
Should we only just keep the pre-defined items that are possible on all platforms and remove the rest? it would be less confusing for users.

Reply:
Let see what @wusyong have to say about that